include/devicetree/clocks.h

```diff
@@ -166,6 +254,32 @@ extern "C" {
  */
 #define DT_CLOCKS_CELL(node_id, cell) DT_CLOCKS_CELL_BY_IDX(node_id, 0, cell)
 
+/**
+ * @brief Equivalent to DT_CLOCKS_HAS_IDX(node_id, idx)
+ * @param inst DT_DRV_COMPAT instance number; may or may not have any clocks property
+ * @param idx index of a clocks property phandle-array whose existence to check
+ * @return 1 if the index exists, 0 otherwise
+ */
+#define DT_INST_CLOCKS_HAS_IDX(inst, idx) \
+	DT_CLOCKS_HAS_IDX(DT_DRV_INST(inst), idx)
+
+/**
+ * @brief Equivalent to DT_CLOCK_HAS_NAME(node_id, name)
+ * @param inst DT_DRV_COMPAT instance number; may or may not have any clock-names property.
+ * @param name lowercase-and-underscores clock-names cell value name to check
+ * @return 1 if the clock name exists, 0 otherwise
+ */
+#define DT_INST_CLOCKS_HAS_NAME(inst, name) \
+	DT_CLOCKS_HAS_NAME(DT_DRV_INST(inst), name)
+
+/**
+ * @brief Equivalent to DT_NUM_CLOCKS(node_id)
```

Comment:
s/node_id/DT_DRV_INST(inst)/, here and above

Reply:
Ok, sure. Done